packages/wagmi/src/hooks/useSendTransaction.ts

```diff
@@ -1,85 +1,32 @@
-import { ErrorCode } from '@ethersproject/logger'
-import { TransactionRequest } from '@ethersproject/providers'
-import { createErrorToast } from '@sushiswap/ui/components/toast'
-import { BigNumber } from 'ethers'
-import { Dispatch, SetStateAction, useCallback, useEffect, useState } from 'react'
-import { ProviderRpcError, usePrepareSendTransaction, useSendTransaction as useSendTransaction_ } from 'wagmi'
-import { SendTransactionArgs, SendTransactionResult } from 'wagmi/actions'
+import { usePrepareSendTransaction, useSendTransaction as useSendTransaction_ } from 'wagmi'
 
-type Args = Parameters<typeof useSendTransaction_>['0']
+export type UsePrepareSendTransactionConfig = Parameters<typeof usePrepareSendTransaction>['0']
+export type UseSendTransactionConfig = Parameters<typeof useSendTransaction_>['0']
 
-export function useSendTransaction({
-  chainId,
-  onError,
-  onMutate,
-  onSuccess,
-  onSettled,
-  prepare,
-  enabled = true,
-  gasMargin = false,
-}: Omit<NonNullable<Args>, 'request' | 'mode'> & {
-  prepare: (request: Dispatch<SetStateAction<(TransactionRequest & { to: string }) | undefined>>) => void
-  enabled?: boolean
+interface UseSendTransaction {
+  prepare?: UsePrepareSendTransactionConfig
+  send?: UseSendTransactionConfig
   gasMargin?: boolean
-}) {
-  const [request, setRequest] = useState<(TransactionRequest & { to: string }) | undefined>()
-  // console.log('useSendTransaction (wrapper) re-runing with', {
-  //   request,
-  //   chainId,
-  //   enabled,
-  // })
-  const { config } = usePrepareSendTransaction({
-    request,
-    chainId,
-    enabled,
-  })
+}
 
-  // console.log('usePrepareSendTransaction returned config', { config })
+export function useSendTransaction({ prepare, send, gasMargin = false }: UseSendTransaction) {
+  const { config } = usePrepareSendTransaction(prepare)
 
-  const _onSettled = useCallback(
-    (
-      data: SendTransactionResult | undefined,
-      e: ProviderRpcError | null,
-      variables: SendTransactionArgs,
-      context: unknown
-    ) => {
-      // console.log('onSettled callback', { data, e, variables, context })
-      // TODO: ignore until wagmi workaround on ethers error
-      // @ts-ignore
-      if (e?.code !== ErrorCode.ACTION_REJECTED) {
-        createErrorToast(e?.message, true)
-      }
+  const gas = gasMargin && config.gas ? (config.gas * 120n) / 100n : config.gas
```

Comment:
I think there's cases where gas margin is being applied more than once

Reply:
or at least potential for this to happen which isn't ideal

Reply:
Don't think it can happen in this case, but in other places it can I think, willfix

Reply:
Yeah, the problem is that it can happen.

Reply:
Don't see how it could happen here